lib/simulator-xcode-9.3.js

```diff
@@ -22,30 +20,45 @@ class SimulatorXcode93 extends SimulatorXcode9 {
+    //   launchd_s 81243 mwakizaka    7u  unix 0x9461828ef425dd69      0t0      ->0x9461828ef27d5021
+    //   launchd_s 81243 mwakizaka    8u  unix 0x9461828ef425b4c9      0t0      /private/tmp/com.apple.launchd.88z8qTMoJA/Listeners
+    //   launchd_s 81243 mwakizaka    9u  unix 0x9461828ef425be29      0t0      /private/tmp/com.apple.launchd.rbqFyGyXrT/com.apple.testmanagerd.unix-domain.socket
+    //   launchd_s 81243 mwakizaka   10u  unix 0x9461828ef425b4c9      0t0      /private/tmp/com.apple.launchd.88z8qTMoJA/Listeners
+    //   launchd_s 81243 mwakizaka   11u  unix 0x9461828ef425c081      0t0      /private/tmp/com.apple.launchd.zHidszZQUZ/com.apple.testmanagerd.remote-automation.unix-domain.socket
+    //   launchd_s 81243 mwakizaka   12u  unix 0x9461828ef425def9      0t0      ->0x9461828ef425de31
+    //   launchd_s 35621 mwakizaka    4u  unix 0x7b7dbedd6d63253f      0t0      /tmp/com.apple.CoreSimulator.SimDevice.150983FD-82FB-4A7B-86DC-D3D264DD90E5/syslogsock
+    //   launchd_s 35621 mwakizaka    5u  unix 0x7b7dbedd6d62f727      0t0      /private/tmp/com.apple.launchd.zuM1XDJcwr/com.apple.webinspectord_sim.socket
+    //   launchd_s 35621 mwakizaka    9u  unix 0x7b7dbedd6d632607      0t0      /private/tmp/com.apple.launchd.KbYwOrA36E/Listeners
+    //   launchd_s 35621 mwakizaka   10u  unix 0x7b7dbedd6d62f727      0t0      /private/tmp/com.apple.launchd.zuM1XDJcwr/com.apple.webinspectord_sim.socket
+    //   launchd_s 35621 mwakizaka   11u  unix 0x7b7dbedd6d62e6bf      0t0      /private/tmp/com.apple.launchd.7wTVfXC9QX/com.apple.testmanagerd.unix-domain.socket
+    //   launchd_s 35621 mwakizaka   12u  unix 0x7b7dbedd6d632607      0t0      /private/tmp/com.apple.launchd.KbYwOrA36E/Listeners
+    //   launchd_s 35621 mwakizaka   13u  unix 0x7b7dbedd6d62e84f      0t0      /private/tmp/com.apple.launchd.g7KQlSsvXT/com.apple.testmanagerd.remote-automation.unix-domain.socket
+    //   launchd_s 35621 mwakizaka   15u  unix 0x7b7dbedd6d62e6bf      0t0      /private/tmp/com.apple.launchd.7wTVfXC9QX/com.apple.testmanagerd.unix-domain.socket
+    //   launchd_s 35621 mwakizaka   16u  unix 0x7b7dbedd6d62e84f      0t0      /private/tmp/com.apple.launchd.g7KQlSsvXT/com.apple.testmanagerd.remote-automation.unix-domain.socket
+    // these _appear_ to always be grouped together by PID for each simulator.
+    // Therefore, by obtaining simulator PID with an expected simulator UDID,
+    // we can get the correct `com.apple.webinspectord_sim.socket`
+    // without depending on the order of `lsof -aUc launchd_sim` result.
+    const {stdout} = await exec('lsof', ['-aUc', 'launchd_sim']);
+    const udidPattern = `([0-9]{1,5}).+${this.udid}`;
+    const udidMatch = stdout.match(new RegExp(udidPattern));
+    if (!udidMatch) {
+      log.debug(`Failed to get Web Inspector socket with ${udidPattern} pattern`);
+      return null;
     }
 
-    return null;
+    const pidPattern = `${udidMatch[1]}.+\\s+(\\S+com.apple.webinspectord_sim.socket)`;
```

Comment:
dots should also be escaped

Reply:
I don't think we need it as ESLint says it is unnecessary.
<img width="1017" alt="Screenshot 2023-01-03 at 16 48 29" src="https://user-images.githubusercontent.com/21286384/210317623-c75f8227-f91f-4bb3-9005-9e56d0889c7c.png">

Reply:
I assume it should be double-backslash

Reply:
sorry, I misunderstood. fixed as https://github.com/appium/appium-ios-simulator/pull/352/commits/eb8f11f9516246bd2f8df80463d567f9fb5b458d